Sources/GRPCHTTP2Core/Client/GRPCClientStreamHandler.swift

```diff
@@ -130,7 +130,17 @@ extension GRPCClientStreamHandler {
         context.fireErrorCaught(error)
       }
 
-    case .ping, .goAway, .priority, .rstStream, .settings, .pushPromise, .windowUpdate,
+    case .rstStream:
+      switch self.stateMachine.unexpectedInboundClose(reason: .streamReset) {
+      case .forwardStatus(let status):
+        context.fireChannelRead(self.wrapInboundOut(.status(status, [:])))
+      case .doNothing:
+        ()
+      case .fireError:
+        assertionFailure("`fireError` should only happen on the server side, never on the client.")
```

Comment:
Not a big deal, but I wonder if we should make this more explicit, e.g. `fireError_clientOnly`

Reply:
Yeah that's a good idea